### ISSUE TITLE
Perfomance: is it possible to speed up?

### ISSUE THREAD

**Issue by rerime** [maildev/maildev#489]
Works horribly slow when > 50k mails.

**Comment by djfarrelly** [comment]
Yes — as the comment notes, this is the UI: it fetches and renders the entire mailbox with no pagination or list virtualization, so it degrades badly at tens of thousands of emails. That's already tracked with a confirmed root-cause analysis in #392.

Closing as a duplicate of #392 — please follow that issue. (The backend REST API does support a `skip` param for windowed fetching; the gap is that the UI doesn't use it and doesn't virtualize the list.)
